WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -108,6 +108,30 @@ def show_user_by_wca_id
     show_user(user)
   end
 
+  def show_registrations_by_comp_id
+    competition = Competition.includes(:events).find_by_id(params[:competition_id])
+    if competition
+      registrations = competition.registrations
+
+      # The request may have a Bearer token, check it
+      current_resource_owner = User.find_by_id(doorkeeper_token&.resource_owner_id)
+      can_manage_competition = current_resource_owner&.can_manage_competition?(competition)
+
+      # If the current user is not authorized or cannot manage the comp,
+      # only display public information: the accepted ones!
```

Comment:
That could be stated in the JSON, hopefully with a better name than `only_public_information: true`.